### PR TITLE
Handle both int and ClientTimeout-based values for Async timeout

### PR DESCRIPTION
**Pull request by DefiDebauchery**
Timeout values are ints, but an attribute `.total` was being called, leading to errors when passing in web3 instances into Call

### CODE REVIEW

**Comment by BobTheBuidler** [comment]
Can you paste a pip list here so I can take a look? I know the .total was necessary for at least one version of aiohttp

**Comment by DefiDebauchery** [comment]
Apologies for the delay ... Interesting; I do see that `ClientTimeout` is an object provided by `aiohttp`, but when I define my timeout in my web3 instance as `request_kwargs` in web3py (per [their docs](https://web3py.readthedocs.io/en/latest/providers.html#web3.providers.rpc.HTTPProvider)), it remains an int.

Perhaps this block of code could be updated to determine whether it's a `ClientTimeout` class? I am definitely not having any issues since changing my local install to be an int.

Anyway, relevant pip list:

```
Package                                           Version
------------------------------------------------- -----------
aiohttp                                           3.8.3
aiohttp-socks                                     0.7.1
eth-abi                                           4.0.0b2
eth-account                                       0.8.0
eth-hash                                          0.5.1
eth-keyfile                                       0.6.0
eth-keys                                          0.4.0
eth-retry                                         0.1.15
eth-rlp                                           0.3.0
eth-typing                                        3.2.0
eth-utils                                         2.1.0
multicall                                         0.7.0
requests                                          2.28.1
web3                                              6.0.0b9
websockets                                        10.4
```

**Comment by BobTheBuidler** [comment]
If you update your PR with an if, else to account for both cases, I can go
ahead and get it merged. What do you think?

**Comment by DefiDebauchery** [comment]
@BobTheBuidler - See updated PR. Format suggestions welcome, if needed.

File: multicall/utils.py [hunk 1]
```diff
@@ -38,8 +39,13 @@ def get_async_w3(w3: Web3) -> Web3:
     if w3 in async_w3s:
         return async_w3s[w3]
     if w3.eth.is_async and isinstance(w3.provider, AsyncBaseProvider):
-        if w3.provider._request_kwargs["timeout"].total < AIOHTTP_TIMEOUT.total:
+        timeout = w3.provider._request_kwargs["timeout"]
+        if isinstance(w3.provider._request_kwargs["timeout"], ClientTimeout):
```
Comment (BobTheBuidler) [inline multicall/utils.py:43]: can just use 'timeout' here

Comment (BobTheBuidler) [reply multicall/utils.py:43]: on 43

**Comment by BobTheBuidler** [comment]
Left one comment to clean up the code a bit, other than that it looks good to me

**Comment by BobTheBuidler** [comment]
I fixed it myself, merging

**Comment by DefiDebauchery** [comment]
Sorry for that. Thanks for the correction!

**Pull request merged**